src/app/store/modules/river-events.js

```diff
@@ -1,45 +1,135 @@
-import actions from '@/app/store/actions'
-import mutations from '@/app/store/mutations'
-import { getReachEvents } from '@/app/services'
+import actions from "@/app/store/actions";
+import mutations from "@/app/store/mutations";
+import { getReachEvents } from "@/app/services";
+import { cloneDeep } from "lodash";
 
 export default {
   namespaced: true,
   state: {
     error: false,
     loading: false,
-    data: null
+    data: null,
   },
   mutations,
   actions: {
     ...actions,
     async getProperty(context, id) {
       try {
-        context.commit('DATA_REQUEST')
-        
-        const result = await getReachEvents(id)
+        context.commit("DATA_REQUEST");
+
+        const result = await getReachEvents(id);
 
         if (!result.errors) {
-          context.commit('DATA_SUCCESS', result)
+          context.commit("DATA_SUCCESS", result);
         }
-
       } catch (error) {
-        context.commit('DATA_ERROR', error)
+        context.commit("DATA_ERROR", error);
       }
-    }
+    },
   },
   getters: {
-    releases: state => {
-      if (state.data) {
-        const results = state.data.filter(item => !!item.data && item.data.length && item.data[0]?.category)
+    // this will return all event dates with the parent event tucked into "event", sorted descending by date.
+    releaseDates: (state, getters) => {
+      let intermediate = [];
+
+      if (getters.releaseEvents?.length) {
+        getters.releaseEvents.forEach((re) => {
+          intermediate = intermediate.concat(
+            re.dates.map((y) => ({ ...y, event: re }))
+          );
+        });
+
+        return cloneDeep(
```

Comment:
I'm not quite sure I understand why you're using `cloneDeep` for every return value here. They don't do that in the Vuex documentation and my impression is that it's not necessary here...

Reply:
It is a way to enforce that parts of state.data aren't getting mutated by any downstream components, it isn't necessary but doesn't cost much. I removed it. I very much doubt that Vuex's documentation would treat the various ways that you can mitigate that type of accident, but they do provide debug logging to console if you mutate state as an option, which is a good alternative.